src/logic/substreams/component.ts

```diff
@@ -86,10 +83,41 @@ export async function createSubstreamsComponent(
     return exitPromise
   }
 
+  // the seconds can be in scientific notation, so we need to convert it to a number
+  function convertScientificNotation(number: number) {
+    const scientificRegex = /^([\d.]+)e([\+,-])(\d+)$/
+    const match = number.toString().match(scientificRegex)
+
+    if (match && match.length === 4) {
+      const base = parseFloat(match[1])
+      const exponent = parseInt(match[2] + match[3])
+      return base * Math.pow(10, exponent)
+    } else {
+      return number
+    }
+  }
+
+  async function ready() {
+    const THRESHOLD = 120 // number of seconds away from head block time
+    const response = await fetch("http://0.0.0.0:9102") // hit prometheus metrics endpoint
+    const metrics = await response.text()
+    const regex = /head_block_time_drift{app="substreams_sink"} ([\d.e+]+)/
+    const match = metrics.match(regex)
+
+    if (match && match[1]) {
+      const headBlockTimeDrift = convertScientificNotation(parseFloat(match[1]))
```

Comment:
Shouldn't `Number(match[1])` be able to handle the scientific notation?
Why is it necessary to parse the match into a float and later process it?

Reply:
absolutely right, the `Number` will handle that for us, great catch. Updated!